snuba/query/extensions.py

```diff
@@ -80,28 +86,40 @@ def get_processor(self) -> ExtensionQueryProcessor:
 }
 
 
-class PerformanceExtension(QueryExtension):
+class PerformanceExtension(QueryExtension[Any]):
     def __init__(self) -> None:
         super().__init__(
             schema=PERFORMANCE_EXTENSION_SCHEMA,
             processor=DummyExtensionProcessor(),
         )
 
+    def parse_payload(cls, payload: Mapping[str, Any]) -> Any:
+        return payload
+
 
-class ProjectExtension(QueryExtension):
+class ProjectExtension(QueryExtension[Any]):
     def __init__(self) -> None:
         super().__init__(
             schema=PROJECT_EXTENSION_SCHEMA,
             processor=DummyExtensionProcessor(),
         )
 
+    def parse_payload(cls, payload: Mapping[str, Any]) -> Any:
+        return payload
+
+
+class TimeSeriesExtensionPayload(TypedDict):
```

Comment:
Why use a `TypedDict` here rather than a dataclass? This should hopefully not be used anywhere other than within the extension itself so compatibility shouldn't be an issue here.